### PR TITLE
api_server: add documentation for public items

### PR DESCRIPTION
**Pull request by dianpopa**
# Reason for This PR

Cross checking an item off [this list](https://github.com/firecracker-microvm/firecracker/issues/609).

## Description of Changes

`[Author TODO: add description of changes.]`

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: src/api_server/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,11 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+
+#![deny(missing_docs)]
+//! Implements the interface for intercepting API requests, forwarding them to the VMM
+//! and responding to the user.
+//! It is constructed on top af an HTTP Server that uses Unix Domain Sockets and `EPOLL` to
```
Comment (berciuliviu) [inline src/api_server/src/lib.rs:7]: on top **of**

File: src/api_server/src/lib.rs [hunk 2]
```diff
@@ -74,23 +84,92 @@ pub struct ApiServer {
...
+    ///     })
+    ///     .unwrap();
+    ///
+    /// thread::sleep(Duration::new(0, 10_000_000));
+    /// let mut sock = UnixStream::connect(PathBuf::from(path_to_socket)).unwrap();
+    /// // Send a GET instance-info request.
+    /// assert!(sock.write_all(b"GET / HTTP/1.1\r\n\r\n").is_ok());
+    /// let mut buf: [u8; 100] = [0; 100];
+    /// assert!(sock.read(&mut buf[..]).unwrap() > 0);
+    /// ```
```
Comment (georgepisaltu) [inline src/api_server/src/lib.rs:172]: Is there a newline between the code example and the next function? I don't know if the GitHub editor is showing it correctly.

Comment (dianpopa) [reply src/api_server/src/lib.rs:172]: There is not a newline. Should there be?

File: src/api_server/src/lib.rs (hunk 1 above)
Comment (dianpopa) [reply src/api_server/src/lib.rs]: Done!

**Review by dianpopa — COMMENTED** [review]
@georgepisaltu @berciuliviu addressed your suggestions. PTAL!

**Review by berciuliviu — APPROVED** [review]
LGTM

File: src/api_server/src/lib.rs (hunk 2 above)
Comment (georgepisaltu) [reply src/api_server/src/lib.rs:172]: I think there should be a newline but it's a nit.

**Review by georgepisaltu — APPROVED** [review]

**Pull request merged**